pkg/route/validation/validation.go

```diff
@@ -342,6 +368,49 @@ func validateWildcardPolicy(host string, policy routev1.WildcardPolicyType, fldP
 	return nil
 }
 
+var (
+	notAllowedHTTPHeaderResponse    = []string{string("Strict-Transport-Security")}
+	notAllowedHTTPHeaderResponseSet = sets.NewString(notAllowedHTTPHeaderResponse...)
+	notAllowedConverters            = []string{"env(", "debug(", "json_query(", "set-var(",
+		"map_str(", "map_str_int(", "map_str_ip(",
+		"map_beg(", "map_beg_int(", "map_end_ip(",
+		"map_sub(", "map_sub_int(", "map_sub_ip(",
+		"map_dir(", "map_dir_int(", "map_dir_ip(",
+		"map_dom(", "map_dom_int(", "map_dom_ip(",
+		"map_end(", "map_end_int(", "map_end_ip(",
+		"map_reg(", "map_reg_int(", "map_reg_ip(",
+		"map_regm(",
+		"map_int(", "map_int_int(", "map_int_ip(",
+		"map_ip(", "map_ip_int(", "map_ip_ip(",
+	}
+	notAllowedConvertersSet = sets.NewString(notAllowedConverters...)
+)
+
+func validateHeaders(key int, value routev1.RouteHTTPHeader, path *field.Path, route routev1.Route) *field.Error {
+	if notAllowedHTTPHeaderResponseSet.Has(value.Name) {
+		return field.NotSupported(path.Index(key).Child("name"), value.Name, []string{fmt.Sprintf("All headers except %v", strings.Join(notAllowedHTTPHeaderResponse, ","))})
+	}
+	if value.Action.Set != nil && value.Action.Type == routev1.Set {
+		if string(value.Action.Set.Value[0]) == "%" && string(value.Action.Set.Value[1]) == "[" && string(value.Action.Set.Value[len(value.Action.Set.Value)-1]) != "]" {
+			msg := fmt.Sprintf("invalid value: The braces are not closed properly.")
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Action.Set.Value, msg)
+		}
```

Comment:
I'm struggling to parse what this is doing? What does a `%[...]` allow me to do here? This hasn't been mentioned in the API at all

Reply:
this is a way to provide dynamic header value.

Reply:
Any links to documentation for this? Is this an HAProxy feature?

Reply:
%[path,map_beg(/etc/hapee-2.6/routes.map,webservers)]

Reply:
http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#7.3